### DISCUSSION TITLE
Mix of 2k displays and 4k: is it possible to set DPI per each?

### DISCUSSION THREAD

**Discussion by vitaly-zdanevich** [i3/i3#5643] [Q&A]
I have read that it is possible in Sway.

**Comment by okraits** [comment]
The DPI value is not set by i3. i3 just uses the DPI value configured with RandR. So the question is rather wether this can be configured with RandR (using xrandr for example).

You might want to have a look at https://wiki.archlinux.org/title/Multihead and `man xrandr`.

**Comment by PMassicotte** [comment]
@vitaly-zdanevich have you found a solution? I am in the same situation.

**Reply by orestisfl** [reply]
See https://ptgamr.notion.site/xrand-4k-mixed-monitors-7a026e5345994267aebfa01a32044a2f and linked articles

**Reply by vitaly-zdanevich** [reply]
I just autostart my Kitty on a high-dpi display with bigger font:

> exec i3-msg 'workspace 1; exec kitty -o font_size=13 --start-as=fullscreen'

and use this display only for terminal apps.

**Comment by PMassicotte** [comment]
Thank you, I will read it right now. My situation is a bit different, as I am using two computers. One with 4K monitors, and another one with 2K monitors. I use the same dotfiles on both, so I would like to find a way to script it so dpi is adjusted to the current resolution. I think this article will provide some hints.

**Comment by PMassicotte** [comment]
Came up with solution (posting in case someone else need)

```bash
#!/bin/bash

# Get the first connected monitor with a resolution
MONITOR_INFO=$(xrandr | grep " connected" | grep -oP '^[^\s]+' | head -n 1)

if [ -z "$MONITOR_INFO" ]; then
    exit 1
fi

# Get the resolution of the selected monitor
RESOLUTION=$(xrandr | grep "^${MONITOR_INFO} connected" | grep -oP '\d+x\d+' | head -n1)

if [ -z "$RESOLUTION" ]; then
    exit 1
fi

# Extract width and height
WIDTH=$(echo $RESOLUTION | cut -d'x' -f1)

# Set the DPI and the cursor size based on resolution
if [ "$WIDTH" -ge 3840 ] || [ "$HEIGHT" -ge 2160 ]; then
    DPI=168
    CURSOR_SIZE=48
elif [ "$WIDTH" -ge 2560 ] || [ "$HEIGHT" -ge 1440 ]; then
    DPI=120
    CURSOR_SIZE=36
else
    DPI=96
fi

# Apply the DPI settings
xrdb -merge <<< "Xft.dpi: $DPI"
xrdb -merge <<< "Xcursor.size: $CURSOR_SIZE"
```

Than in i3 `config` I used

```
exec_always  ~/.config/set_dpi.sh
```
